### DISCUSSION TITLE
Update gatsby 2v to v4

### DISCUSSION THREAD

**Discussion by PercyTuncar** [gatsbyjs/gatsby#35568] [Help]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

I am migrating from Gatsby v2 to V4 but when doing Gatsby Develop I get an error 

### Reproduction Link

https://github.com/PercyTuncar/bug-repro-minimal.git

### Steps to Reproduce

1.  package.json
2. gatsby-config.js
3.
...

### Expected Result

Change the version to be able to deploy

### Actual Result

$ gatsby develop
success compile gatsby files - 5.671s
success load gatsby config - 1.242s
warn Plugin gatsby-plugin-react-helmet is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.0.0     
warn Plugin gatsby-transformer-yaml is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.0.15       
warn Plugin gatsby-plugin-offline is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.4.0
warn Plugin gatsby-transformer-remark is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.12.0     
warn Plugin gatsby-transformer-remark is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.12.0     
warn Plugin gatsby-plugin-offline is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.4.0          
warn Plugin gatsby-transformer-remark is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.12.0     
warn Plugin gatsby-plugin-google-tagmanager is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.0.0
warn Plugin gatsby-transformer-remark is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.12.0     
warn Plugin gatsby-plugin-manifest is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.25.0        
⠋ load plugins
'WMIC' is not recognized as an internal or external command,
warn Plugin gatsby-source-filesystem is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.2.0       
warn Plugin gatsby-source-filesystem is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.2.0       
warn Plugin gatsby-transformer-sharp is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.12.1      
warn Plugin gatsby-transformer-sharp is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.12.1      
warn Plugin gatsby-transformer-remark is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.12.0     
warn Plugin gatsby-plugin-sass is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.0.0             
warn Plugin gatsby-plugin-postcss is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.0.0
warn Plugin gatsby-plugin-nprogress is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.0.0        
warn Plugin gatsby-plugin-sitemap is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.0.0          
warn Plugin gatsby-plugin-netlify-cms is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.0.101    
warn Plugin gatsby-plugin-netlify is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.0.0
warn Warning: there are unknown plugin options for "gatsby-plugin-manifest": display_override
Please open an issue at https://ghub.io/gatsby-plugin-manifest if you believe this option is valid.
warn Plugin gatsby-remark-images is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.0.0           
warn Plugin gatsby-remark-responsive-iframe is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.0.0
warn Plugin gatsby-plugin-react-helmet is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.0.0     
warn Plugin gatsby-transformer-yaml is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.0.15       
warn Plugin gatsby-plugin-offline is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.4.0          
warn Plugin gatsby-transformer-remark is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.12.0     
warn Plugin gatsby-transformer-remark is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.12.0     
warn Plugin gatsby-plugin-offline is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.4.0          
warn Plugin gatsby-transformer-remark is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.12.0     
warn Plugin gatsby-plugin-google-tagmanager is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.0.0
warn Plugin gatsby-transformer-remark is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.12.0     
warn Plugin gatsby-plugin-manifest is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.25.0        
warn Plugin gatsby-source-filesystem is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.2.0       
warn Plugin gatsby-source-filesystem is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.2.0       
warn Plugin gatsby-transformer-sharp is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.12.1      
warn Plugin gatsby-transformer-sharp is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.12.1      
warn Plugin gatsby-remark-images is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.0.0
warn Plugin gatsby-remark-responsive-iframe is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.0.0
warn Plugin gatsby-transformer-remark is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.12.0     
warn Plugin gatsby-plugin-sass is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.0.0             
warn Plugin gatsby-plugin-postcss is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.0.0          
warn Plugin gatsby-plugin-nprogress is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.0.0        
warn Plugin gatsby-plugin-sitemap is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.0.0
warn Plugin gatsby-plugin-netlify-cms is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.0.101    
warn Plugin gatsby-plugin-netlify is not compatible with your gatsby version 4.13.1 - It requires gatsby@^2.0.0          

 ERROR 

Error in "C:\Users\percy\Desktop\Web Apps\codenightla\node_modules\gatsby-plugin-sharp\gatsby-node.js":
Something went wrong installing the "sharp" module

The specified procedure could not be found.
\\?\C:\Users\percy\Desktop\Web Apps\codenightla\node_modules\sharp\build\Release\sharp-win32-x64.node

Possible solutions:
- Install with the --verbose flag and look for errors: "npm install --ignore-scripts=false --verbose sharp"
- Install for the current win32-x64 runtime: "npm install --platform=win32 --arch=x64 sharp"
- Consult the installation documentation: https://sharp.pixelplumbing.com/install
- Ensure the version of sharp aligns with the gatsby-plugin-manifest package: "npm ls sharp"

  Error: Something went wrong installing the "sharp" module
  The specified procedure could not be found. 
  \\?\C:\Users\percy\Desktop\Web Apps\codenightla\node_modules\sharp\build\Release\sharp-win32-x64.node
  Possible solutions:
  - Install with the --verbose flag and look for errors: "npm install --ignore-scripts=false --verbose sharp"
  - Install for the current win32-x64 runtime: "npm install --platform=win32 --arch=x64 sharp"
  - Consult the installation documentation: https://sharp.pixelplumbing.com/install
  - Ensure the version of sharp aligns with the gatsby-plugin-manifest package: "npm ls sharp"

  - sharp.js:31 Object.<anonymous>
    [codenightla]/[sharp]/lib/sharp.js:31:9

  - loader.js:1085 Module._compile
    internal/modules/cjs/loader.js:1085:14

  - loader.js:1114 Object.Module._extensions..js
    internal/modules/cjs/loader.js:1114:10

  - loader.js:950 Module.load
    internal/modules/cjs/loader.js:950:32

  - helpers.js:92 require
    internal/modules/cjs/helpers.js:92:18

  - constructor.js:8 Object.<anonymous>
    [codenightla]/[sharp]/lib/constructor.js:8:1

not finished load plugins - 13.478s

![0111](https://user-images.githubusercontent.com/33427262/166192890-7c45669c-a27e-4769-8c83-8131845a1739.png)
![022](https://user-images.githubusercontent.com/33427262/166192887-1918ef6b-a91c-4570-be30-a51be27ae0d7.png)

### Environment

```shell
node --version
v14.17.3

npm --version
8.5.1

gatsby --version
Gatsby CLI version: 4.12.1
Gatsby version: 4.13.1
```

### Config Flags

none

**Comment by LekoArts** [comment]
You'll need to update all packages, not only `gatsby`
